### ISSUE TITLE
Shadow mapping crashes inside graphics driver (Mesa)

### ISSUE THREAD

**Issue by Lazerbeak12345** [minetest/minetest#12737]
This issue was first reported here. https://github.com/minetest-whynot/whynot-game/issues/126 We've reduced it down to be an issue with `minetest_game` - if not an engine bug.

---

## Description of what happens

1. Use the `minetest.conf` file below in the resources section for your client settings. THIS BUG DOESN'T HAPPEN WHEN USING THE DEFAULT `minetest.conf` FILE
2. Open a world on the minetest game commit [9528c0f8b93d6934930e99c3c116df275fb0e4bc](https://github.com/minetest/minetest_game/commit/9528c0f8b93d6934930e99c3c116df275fb0e4bc) or later without any additional mods. (though it still happens with mods)
3. Once it "loads" all I can see is grey, and the hud. Sometimes it's the vanilla hud, but in this screenshot, no. (screenshot was from the Whynot issue but the same grey screen still happens and the hud behavior is identical). The user cannot interact with chat, and doesn't see any chat come in. ![image](https://user-images.githubusercontent.com/22641188/185767865-bd6a9af1-acbf-4571-b537-8c5c12bdd48b.png)
4. Exits after a few seconds. (with unix error code 134)
5. The logs seem to hold no error information.

This bug doesn't happen on older commits

## Resources

Backtrace: https://github.com/minetest/minetest/issues/12737#issuecomment-1236169258

<details><summary>git bisect result</summary>
<p>

```
9528c0f8b93d6934930e99c3c116df275fb0e4bc is the first bad commit
commit 9528c0f8b93d6934930e99c3c116df275fb0e4bc
Author: lhofhansl <larsh@apache.org>
Date:   Sat Apr 2 00:35:36 2022 -0700

    Adjust shadow intensity based on cloud density (#2938)

 minetest.conf.example |  2 +-
 mods/weather/init.lua | 16 +++++++++++++---
 settingtypes.txt      |  2 +-
 3 files changed, 15 insertions(+), 5 deletions(-)
```

</p></details>

<details><summary>logs</summary>
<p>

```
$ minetest --worldname "testbug" --go
2022-09-01 12:53:50: [Main]: Using world specified by --worldname on t
he command line
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation n
ot permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation n
ot permitted (1)
         __.               __.                 __.  
  _____ |__| ____   _____ /  |_  _____  _____ /  |_ 
 /     \|  |/    \ /  __ \    _\/  __ \/   __>    _\
|  Y Y  \  |   |  \   ___/|  | |   ___/\___  \|  |  
|__|_|  /  |___|  /\______>  |  \______>_____/|  |  
      \/ \/     \/         \/                  \/   
2022-09-01 12:53:51: ACTION[Main]: World at [/home/nate/.minetest/worl
ds/testbug]
2022-09-01 12:53:51: ACTION[Main]: Server for gameid="minetest_game_gi
t" listening on 0.0.0.0:50633.
2022-09-01 12:53:52: ACTION[Server]: Moving singleplayer to static spa
wnpoint at (60,4,0)
2022-09-01 12:53:52: ACTION[Server]: singleplayer [127.0.0.1] joins ga
me. List of players: singleplayer
fish: Job 1, 'minetest --worldname "testbug"…' terminated by signal SI
GABRT (Abort)
```

</p></details>

<details><summary><em>minimal</em> minetest.conf file for client</summary>
<p>

```
shadow_map_color = true
enable_dynamic_shadows = true

# 	If this value is too low, the bug doesn't happen.
# 	The default value is now 2048 (which does not crash),
# but below is what I found on my computer. I played around
# with it quite a bit to try and find the lowest possible value
# that this fails on, but it fluctuates - perhaps with ram use?
shadow_map_texture_size = 8192
```

</p></details>

## System Info

Minetest version `5.6.0`

```
Operating System: Fedora Linux 36
KDE Plasma Version: 5.25.4
KDE Frameworks Version: 5.97.0
Qt Version: 5.15.5
Kernel Version: 5.19.4-200.fc36.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 4 × Intel® Core™ i5-2520M CPU @ 2.50GHz
Memory: 7.6 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 3000
Manufacturer: LENOVO
Product Name: 4180BV1
System Version: ThinkPad T420
```

This bug happens on both X11 and Wayland, on both KDE and Gnome.

## TODOS for myself involving this issue.

- use `git bisect` on minetest itself to determine the first commit this happens on.

**Comment by sfan5** [comment]
Can you produce a backtrace?

**Comment by Lazerbeak12345** [comment]
Not sure how. (before this was transferred I didn't have the knowledge before hand on how to report an engine bug or what would qualify)

**Comment by sfan5** [comment]
`gdb -q -batch -ex 'set confirm off' -ex 'r' -ex 'bt' --args minetest --worldname "testbug" --go`
To get meaningful results you'll want a debug build, your distro might have a debug package you can install if you're lucky.

**Comment by PrairieAstronomer** [comment]
The grey screen, I have seen that, mostly on servers upon connecting. If you wait long enough, and try to chat, it will usually go away. As for crashing, it never crashes for me.

**Comment by Lazerbeak12345** [comment]
> The grey screen, I have seen that, mostly on servers upon connecting. If you wait long enough, and try to chat, it will usually go away. As for crashing, it never crashes for me. It usually is completely grey, except for chat messages.

I think I've had that happen years ago - but this bug is _not_ the same.

- The chat feature doesn't work at all in my case.
- Waiting 100% _always_ causes a crash.

**Comment by Lazerbeak12345** [comment]
> `gdb -q -batch -ex 'set confirm off' -ex 'r' -ex 'bt' --args minetest --worldname "testbug" --go` To get meaningful results you'll want a debug build, your distro might have a debug package you can install if you're lucky.

@sfan5

Sorry for the delay. I'm quite busy recently.

Would this be useful at all for you without a debug build?

**Comment by Lazerbeak12345** [comment]
Just followed the instructions and built minetest from source on 2854c19792a1c11c46788d15b746cf3762984a38

```
$ gdb -q -batch -ex 'set confirm off' -ex 'r' -ex 'bt' --args ./bin/minetest --worldname "test" --go
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
2022-09-03 11:31:44: [Main]: Using world specified by --worldname on the command line
[New Thread 0x7fffe8547640 (LWP 43262)]
[Thread 0x7fffe8547640 (LWP 43262) exited]
[New Thread 0x7fffe8547640 (LWP 43265)]
[Thread 0x7fffe8547640 (LWP 43265) exited]
[New Thread 0x7fffe8547640 (LWP 43266)]
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[New Thread 0x7fffe7a4f640 (LWP 43267)]
[Thread 0x7fffe7a4f640 (LWP 43267) exited]
[New Thread 0x7fffe7a4f640 (LWP 43270)]
[New Thread 0x7fffe720d640 (LWP 43271)]
[New Thread 0x7fffe6a0c640 (LWP 43274)]
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[New Thread 0x7fffe620b640 (LWP 43275)]
[New Thread 0x7fffe5623640 (LWP 43276)]
[New Thread 0x7fffe4dd4640 (LWP 43281)]
[New Thread 0x7fffc75e0640 (LWP 43284)]
[New Thread 0x7fffc6ddf640 (LWP 43285)]
[New Thread 0x7fffc52cb640 (LWP 43289)]
[New Thread 0x7fffc4aca640 (LWP 43290)]
         __.               __.                 __.  
  _____ |__| ____   _____ /  |_  _____  _____ /  |_ 
 /     \|  |/    \ /  __ \    _\/  __ \/   __>    _\
|  Y Y  \  |   |  \   ___/|  | |   ___/\___  \|  |  
|__|_|  /  |___|  /\______>  |  \______>_____/|  |  
      \/ \/     \/         \/                  \/   
2022-09-03 11:31:46: ACTION[Main]: World at [/home/nate/projects/minetest/bin/../worlds/test]
2022-09-03 11:31:46: ACTION[Main]: Server for gameid="minetest" listening on 0.0.0.0:50205.
[New Thread 0x7fffb7fff640 (LWP 43291)]
[New Thread 0x7fffb77fe640 (LWP 43292)]
[New Thread 0x7fffb6ffd640 (LWP 43293)]
[New Thread 0x7fffb62fc640 (LWP 43294)]
[New Thread 0x7fffb51c8640 (LWP 43295)]
2022-09-03 11:31:48: ACTION[Server]: singleplayer [127.0.0.1] joins game. List of players: singleplayer

Thread 11 "minetest:gdrv0" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffe4dd4640 (LWP 43281)]
0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#0  0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007ffff6e3e9c6 in raise () from /lib64/libc.so.6
#2  0x00007ffff6e287f4 in abort () from /lib64/libc.so.6
#3  0x00007fffd66a659f in _crocus_batch_flush.cold () from /usr/lib64/dri/crocus_dri.so
#4  0x00007fffd76209c2 in crocus_draw_vbo () from /usr/lib64/dri/crocus_dri.so
#5  0x00007fffd6c0a7d6 in tc_call_draw_single () from /usr/lib64/dri/crocus_dri.so
#6  0x00007fffd6c0a3b9 in tc_batch_execute () from /usr/lib64/dri/crocus_dri.so
#7  0x00007fffd66c2d04 in util_queue_thread_func () from /usr/lib64/dri/crocus_dri.so
#8  0x00007fffd66c288b in impl_thrd_routine () from /usr/lib64/dri/crocus_dri.so
#9  0x00007ffff6e8ce2d in start_thread () from /lib64/libc.so.6
#10 0x00007ffff6f121b0 in clone3 () from /lib64/libc.so.6
```

**Comment by Lazerbeak12345** [comment]
@sfan5 is this what you were looking for?

**Comment by SmallJoker** [comment]
@Lazerbeak12345 Look up the setting `enable_vbo` and change it to `false`. Restart Minetest afterwards. Does it fix your issue?

Either way this looks a lot like a driver bug.

**Comment by Lazerbeak12345** [comment]
I just want to link this here.... ABRT automatically makes this upon this type of crash for RHL based systems like mine (Fedora)

https://retrace.fedoraproject.org/faf/reports/325988/

**Comment by Lazerbeak12345** [comment]
> @Lazerbeak12345 Look up the setting `enable_vbo` in Minetest and change it to `false`. Restart Minetest afterwards. Does it fix your issue?

@SmallJoker 

No, it still crashes.

**Comment by Lazerbeak12345** [comment]
Updated OP with link to backtrace in resources, and removed relevant TODOs

**Comment by savilli** [comment]
Try to disable crocus driver and see if it crashes again. `LIBVA_DRIVER_NAME=i965 minetest` command.

**Comment by Desour** [comment]
>2. Open a world on the minetest game commit [9528c0f8b93d6934930e99c3c116df275fb0e4bc](https://github.com/minetest/minetest_game/commit/9528c0f8b93d6934930e99c3c116df275fb0e4bc) or later without any additional mods.

Have you tried turning shadow mapping off?

**Comment by Lazerbeak12345** [comment]
> Have you tried turning shadow mapping off?

There's a lot of shadow mapping settings. None of them look like an "off." Which setting(s) should be set to what value?

**Comment by Lazerbeak12345** [comment]
> Try to disable crocus driver and see if it crashes again. `LIBVA_DRIVER_NAME=i965 minetest` command.

Thanks for the idea.... I just tried that command (in bash, to be sure we are on the same page here) and the error still happened.

**Comment by Lazerbeak12345** [comment]
Added (perhaps too agressive) clarification that when using the default (ie remove the file) `minetest.conf` file, the bug does not happen. One of the settings I've changed in my config file (which used to work in 5.5) does not work in 5.6

**Comment by sfan5** [comment]
It looks like the crash inside that graphics driver also affects other applications, so I'd say it's very likely this isn't Minetest's fault.

https://retrace.fedoraproject.org/faf/problems/bthash/?bth=5465d0bebbda7a4b05ae08780a4d683c43bb1bb9&bth=82516cb1f57e3d39839de4db9c79e2a456eff98a&bth=4ba7968af8fb9692c4b147a0cdd0a316c5705361&bth=b4f341e65c3da44b8f1520a74a59e4930f02be6a&bth=e0cda6bfd10da21d985a83ff17930e166387774d&bth=66051c9b6bd4b77cf7db4e68eb50f5c29a2100ba&bth=a2962a38d23cc9aa2640ca1cdc14b49a3320d80d&bth=a10558f49b898cc996bc36bffbc15141dbfdef01&bth=a83fb4d27196211e4cabae888bfb778b121bf4de&bth=63e136f4a3acba9477ad2aafadd42877707bd089

**Comment by Desour** [comment]
> There's a lot of shadow mapping settings. None of them look like a setting with an "off." Which setting(s) should be set to what value?

`enable_dynamic_shadows = false`
(You can also try using lower quality shadows afterwards.)

**Comment by Lazerbeak12345** [comment]
I actually have more info on this. I might be able to get around to posting it later today.

The bug is for sure related to dynamic shadows, and most likely in relation to their interactions with the graphics driver, mesa.

**Comment by Lazerbeak12345** [comment]
Updated the `minetest.conf` in the resources section. (see there for more details)

**Comment by Lazerbeak12345** [comment]
TLDR for that conf file though... @Desour was spot-on.

**Comment by savilli** [comment]
> > Try to disable crocus driver and see if it crashes again. `LIBVA_DRIVER_NAME=i965 minetest` command.
> 
> Thanks for the idea.... I just tried that command (in bash, to be sure we are on the same page here) and the error still happened.

Did it crash with the crocus driver in backtrace?

**Comment by HybridDog** [comment]
> `LIBVA_DRIVER_NAME=i965 minetest`

I think [LIBVA_DRIVER_NAME](https://wiki.archlinux.org/title/Hardware_video_acceleration#Configuring_VA-API) affects VA-API hardware video en- and decoding, which is unrelated to Minetest. 

If your system supports Vulkan, you could test if it crashes with Zink (OpenGL implemented in Vulkan): `MESA_LOADER_DRIVER_OVERRIDE=zink minetest`
The supported drivers for [MESA_LOADER_DRIVER_OVERRIDE](https://docs.mesa3d.org/envvars.html) can be found with `find /usr/lib/dri -name '*_dri.so'` as far as I know.
You could also check if it crashes even with software rendering: `LIBGL_ALWAYS_SOFTWARE=1 minetest`.
There also exist some environment variables for debugging, e.g. `export MESA_DEBUG=1; export LIBGL_DEBUG=verbose`

**Comment by Lazerbeak12345** [comment]
> > > Try to disable crocus driver and see if it crashes again. `LIBVA_DRIVER_NAME=i965 minetest` command.
> > 
> > 
> > Thanks for the idea.... I just tried that command (in bash, to be sure we are on the same page here) and the error still happened.
> 
> Did it crash with the crocus driver in backtrace?

@savilli  Yes, it was verbatim the same as the earlier posted backtrace.

**Comment by Lazerbeak12345** [comment]
> If your system supports Vulkan, you could test if it crashes with Zink (OpenGL implemented in Vulkan): `MESA_LOADER_DRIVER_OVERRIDE=zink minetest`

@HybridDog 

Here's the collapsed backtrace for that command (in bash, again)

<details><summary>backtrace using zink</summary>
<p>

```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7fffe8547640 (LWP 11875)]
[Thread 0x7fffe8547640 (LWP 11875) exited]
[New Thread 0x7fffe8547640 (LWP 11878)]
[Thread 0x7fffe8547640 (LWP 11878) exited]
[New Thread 0x7fffe8547640 (LWP 11879)]
[New Thread 0x7fffe7a4f640 (LWP 11880)]
[Thread 0x7fffe7a4f640 (LWP 11880) exited]
[New Thread 0x7fffe7a4f640 (LWP 11883)]
[New Thread 0x7fffe720d640 (LWP 11884)]
[New Thread 0x7fffe6a0c640 (LWP 11887)]
[New Thread 0x7fffe620b640 (LWP 11888)]

Thread 1 "minetest" received signal SIGABRT, Aborted.
0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#0  0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007ffff6e3e9c6 in raise () from /lib64/libc.so.6
#2  0x00007ffff6e287f4 in abort () from /lib64/libc.so.6
#3  0x00007ffff72a2b57 in __gnu_cxx::__verbose_terminate_handler() [clone .cold] () from /lib64/libstdc++.so.6
#4  0x00007ffff72ae3dc in __cxxabiv1::__terminate(void (*)()) () from /lib64/libstdc++.so.6
#5  0x00007ffff72ae447 in std::terminate() () from /lib64/libstdc++.so.6
#6  0x00007ffff72ae6a8 in __cxa_throw () from /lib64/libstdc++.so.6
#7  0x00007ffff72a53f6 in std::__throw_logic_error(char const*) () from /lib64/libstdc++.so.6
#8  0x000000000041a02d in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string (this=0x7fffffff3f10, __s=0x0, __a=...) at /usr/include/c++/12/bits/basic_string.h:639
#9  0x0000000000a30cc3 in OpenGLProcedures::LoadAllProcedures (this=0xfff240 <GL>, cmgr=0x1083760) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/mt_opengl_loader.cpp:760
#10 0x0000000000a982c8 in irr::video::COpenGLDriver::initDriver (this=0x1320e20) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/COpenGLDriver.cpp:54
#11 0x0000000000aa5ec6 in irr::video::createOpenGLDriver (params=..., io=0x10873a0, contextManager=0x1083760) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/COpenGLDriver.cpp:4223
#12 0x0000000000a35f3e in irr::CIrrDeviceLinux::createDriver (this=0x1303000) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/CIrrDeviceLinux.cpp:541
#13 0x0000000000a34a0c in irr::CIrrDeviceLinux::CIrrDeviceLinux (this=0x1303000, param=..., __in_chrg=<optimized out>, __vtt_parm=<optimized out>) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/CIrrDeviceLinux.cpp:165
#14 0x0000000000a33f36 in irr::createDeviceEx (params=...) at /home/nate/projects/minetest/lib/irrlichtmt/source/Irrlicht/Irrlicht.cpp:80
#15 0x000000000055a432 in RenderingEngine::RenderingEngine (this=0x1302fa0, receiver=0x107f5a0) at /home/nate/projects/minetest/src/client/renderingengine.cpp:144
#16 0x0000000000469ebc in ClientLauncher::init_engine (this=0x7fffffffdbc0) at /home/nate/projects/minetest/src/client/clientlauncher.cpp:339
#17 0x0000000000468893 in ClientLauncher::run (this=0x7fffffffdbc0, start_data=..., cmd_args=...) at /home/nate/projects/minetest/src/client/clientlauncher.cpp:104
#18 0x00000000009259c8 in main (argc=4, argv=0x7fffffffdd38) at /home/nate/projects/minetest/src/main.cpp:256
```

</p></details>

**Comment by Lazerbeak12345** [comment]
> The supported drivers for [MESA_LOADER_DRIVER_OVERRIDE](https://docs.mesa3d.org/envvars.html) can be found with `find /usr/lib/dri -name '*_dri.so'` as far as I know.

Here's the result of that command:

```
/usr/lib/dri/r300_dri.so
/usr/lib/dri/crocus_dri.so
/usr/lib/dri/iris_dri.so
/usr/lib/dri/i915_dri.so
/usr/lib/dri/nouveau_dri.so
/usr/lib/dri/virtio_gpu_dri.so
/usr/lib/dri/vmwgfx_dri.so
/usr/lib/dri/radeonsi_dri.so
/usr/lib/dri/kms_swrast_dri.so
/usr/lib/dri/swrast_dri.so
/usr/lib/dri/zink_dri.so
/usr/lib/dri/r600_dri.so
```

**Comment by Lazerbeak12345** [comment]
> You could also check if it crashes even with software rendering: `LIBGL_ALWAYS_SOFTWARE=1 minetest`.

It did not crash (but was slow, as expected)

**Comment by Lazerbeak12345** [comment]
> There exist some environment variables for debugging, which might be helpful, e.g. `export MESA_DEBUG=1; export LIBGL_DEBUG=verbose`.

Here's the (collapsed) result of exporting those variables before running.

<details><summary>mesa debug and libgl debug active, default driver.</summary><p>

```
gdb -q -batch -ex 'set confirm off' -ex 'r' -ex 'bt' --args ./bin/minetest --worldname "test" --go
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
2022-09-05 18:58:31: [Main]: Using world specified by --worldname on the command line
[New Thread 0x7fffe8547640 (LWP 12701)]
[Thread 0x7fffe8547640 (LWP 12701) exited]
[New Thread 0x7fffe8547640 (LWP 12704)]
[Thread 0x7fffe8547640 (LWP 12704) exited]
[New Thread 0x7fffe8547640 (LWP 12705)]
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[New Thread 0x7fffe7a4f640 (LWP 12706)]
[Thread 0x7fffe7a4f640 (LWP 12706) exited]
[New Thread 0x7fffe7a4f640 (LWP 12709)]
[New Thread 0x7fffe720d640 (LWP 12710)]
[New Thread 0x7fffe6a0c640 (LWP 12713)]
[New Thread 0x7fffe620b640 (LWP 12714)]
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/nate/.drirc: No such file or directory.
libGL: using driver i915 for 17
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/nate/.drirc: No such file or directory.
libGL: pci id for fd 17: 8086:0126, driver crocus
libGL: MESA-LOADER: dlopen(/usr/lib64/dri/crocus_dri.so)
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/nate/.drirc: No such file or directory.
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/nate/.drirc: No such file or directory.
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /home/nate/.drirc: No such file or directory.
[New Thread 0x7fffe5623640 (LWP 12717)]
libGL: Using DRI3 for screen 0
[New Thread 0x7fffe4dd4640 (LWP 12719)]
[New Thread 0x7fffc7754640 (LWP 12723)]
[New Thread 0x7fffc6f53640 (LWP 12724)]
[New Thread 0x7fffc54b3640 (LWP 12727)]
[New Thread 0x7fffc4cb2640 (LWP 12728)]
         __.               __.                 __.  
  _____ |__| ____   _____ /  |_  _____  _____ /  |_ 
 /     \|  |/    \ /  __ \    _\/  __ \/   __>    _\
|  Y Y  \  |   |  \   ___/|  | |   ___/\___  \|  |  
|__|_|  /  |___|  /\______>  |  \______>_____/|  |  
      \/ \/     \/         \/                  \/   
2022-09-05 18:58:33: ACTION[Main]: World at [/home/nate/projects/minetest/bin/../worlds/test]
2022-09-05 18:58:33: ACTION[Main]: Server for gameid="minetest" listening on 0.0.0.0:59546.
[New Thread 0x7fffb7fff640 (LWP 12729)]
[New Thread 0x7fffb77fe640 (LWP 12730)]
[New Thread 0x7fffb6ffd640 (LWP 12731)]
[New Thread 0x7fffb64fc640 (LWP 12732)]
[New Thread 0x7fffb5406640 (LWP 12737)]
2022-09-05 18:58:34: ACTION[Server]: singleplayer [127.0.0.1] joins game. List of players: singleplayer

Thread 11 "minetest:gdrv0" received signal SIGABRT, Aborted.
[Switching to Thread 0x7fffe4dd4640 (LWP 12719)]
0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#0  0x00007ffff6e8ec4c in __pthread_kill_implementation () from /lib64/libc.so.6
#1  0x00007ffff6e3e9c6 in raise () from /lib64/libc.so.6
#2  0x00007ffff6e287f4 in abort () from /lib64/libc.so.6
#3  0x00007fffd66a659f in _crocus_batch_flush.cold () from /usr/lib64/dri/crocus_dri.so
#4  0x00007fffd76209c2 in crocus_draw_vbo () from /usr/lib64/dri/crocus_dri.so
#5  0x00007fffd6c0a7d6 in tc_call_draw_single () from /usr/lib64/dri/crocus_dri.so
#6  0x00007fffd6c0a3b9 in tc_batch_execute () from /usr/lib64/dri/crocus_dri.so
#7  0x00007fffd66c2d04 in util_queue_thread_func () from /usr/lib64/dri/crocus_dri.so
#8  0x00007fffd66c288b in impl_thrd_routine () from /usr/lib64/dri/crocus_dri.so
#9  0x00007ffff6e8ce2d in start_thread () from /lib64/libc.so.6
#10 0x00007ffff6f121b0 in clone3 () from /lib64/libc.so.6
```

</p></details>

There's very little added.

**Comment by sfan5** [comment]
What would make sense for you to try is the i965 driver, because it's the predecessor of crocus and supports the same hardware. You don't have it installed however.
In any case this is a driver bug and not something we can solve.

**Comment by Desour** [comment]
> > If your system supports Vulkan, you could test if it crashes with Zink (OpenGL implemented in Vulkan): `MESA_LOADER_DRIVER_OVERRIDE=zink minetest`
> 
> Here's the collapsed backtrace for that command (in bash, again)
> backtrace using zink
> 
> This backtrace is _very_ different.

FYI, it seems like the string constructor got a NULL, so `glGetString(EXTENSION)` failed (see <https://docs.gl/gl2/glGetString>).
GL >= 3 actually uses `glGetStringi` (and `glGetIntegerv` with `NUM_EXTENSIONS`) for extensions. (And we are requesting GL 3.0 for X, AFAIK.)

**Comment by Lazerbeak12345** [comment]
I'm unsure what the next steps are. I'll run the full debug report for the bug thing I linked earlier. I'll also link this issue, but I don't know if this is the correct way of doing this.

**Comment by savilli** [comment]
If you're using an unstable linux distro like Fedora, you should be used to deal with its problems.
You can try to use a different mesa driver or report the problem to mesa devs.
